src/api-service/__app__/onefuzzlib/request_access.py

```diff
@@ -0,0 +1,115 @@
+from typing import Dict, List
+from uuid import UUID
+
+from onefuzztypes.models import ApiAccessRule
+from pydantic import BaseModel, parse_raw_as
+
+
+class RuleConflictError(Exception):
+    def __init__(self, message: str) -> None:
+        super(RuleConflictError, self).__init__(message)
+        self.message = message
+
+
+class RequestAccess:
+    """
+    Stores the rules associated with a the request paths
+    """
+
+    class Rules:
+        allowed_groups_ids: List[UUID]
```

Comment:
User and group IDs are stored in the same list ?

Reply:
yes, might differentiate depending on the usage but that will be done in the PR that uses this class #1413